packages/core/src/performers/step-performer/StepPerformerPromptCreator.ts

```diff
@@ -196,15 +197,22 @@ export class StepPerformerPromptCreator {
       "throw new Error(\"Unable to find the 'Submit' button element in the current context.\");",
       "```",
       "",
+      "#### Example of providing the view hierarchy snippet:",
+      "<VIEW_HIERARCHY_SNIPPET>",
+      `<RCTViewComponentView alpha="1.0" class="RCTViewComponentView" focused="false" height="769" id="detox_temp_0_0_0_0_0_0_0_0_0_0_0_0" label="User ID: 12345" tag="124" visibility="visible" width="393" x="0" y="0">`,
+      `<RCTTextView alpha="1.0" class="RCTTextView" focused="false" height="20" id="detox_temp_0_0_0_0_0_0_0_0_0_0_0_0_1" label="User ID: 12345" visibility="visible" width="393" x="0" y="0"/>`,
+      `<UIButton alpha="1.0" class="UIButton" focused="false" height="50" id="user-profile-12345" visibility="visible" width="200" text="User Profile"/>`,
+      `</RCTViewComponentView>`,
+      "</VIEW_HIERARCHY_SNIPPET>",
```

Comment:
View hierarchy vs. Pre-code-matcher
```ts
<CACHE_VALIDATION_MATCHER>
await element(by.id('user-1234')).toExist();
</CACHE_VALIDATION_MATCHER>
```

Reply:
We should ask Pilot, to generate the necessary element existence checks for all elements that he's going to interact with in the step code.

Reply:
agreed face-to-face, let's try this approach 👑